src/main/java/frc/robot/note_tracking/NoteTrackingManager.java

```diff
@@ -355,18 +355,42 @@ public boolean mapContainsNote() {
   }
 
   public void addNoteToMap(Translation2d pose) {
-    noteMap.add(new NoteMapElement(Timer.getFPGATimestamp() + NOTE_MAP_LIFETIME, pose));
+    noteMap.add(new NoteMapElement(Timer.getFPGATimestamp() + NOTE_MAP_LIFETIME, pose, 10));
+  }
+
+  private NoteMapElement updatedHealth(NoteMapElement oldElement) {
+    return new NoteMapElement(
+        oldElement.expiresAt(), oldElement.noteTranslation(), oldElement.health() - 1);
   }
 
   private void updateMap() {
     List<Pose2d> visionNotes = getFilteredNotePoses();
 
+    var filteredNotesInBox =
+        noteMap.stream()
+            .filter(
+                element -> {
+                  return (RobotConfig.get().perfToggles().noteMapBoundingBox()
+                      && safeToTrack()
+                      && noteInView(element.noteTranslation()));
+                });
+
+    filteredNotesInBox.forEach(
+        element -> {
+          noteMap.remove(element);
+        });
+
+    var updatedHealthNotes = filteredNotesInBox.map(this::updatedHealth).toList();
+
+    updatedHealthNotes.stream()
+        .forEach(
+            element -> {
+              noteMap.add(element);
+            });
+
     noteMap.removeIf(
         element -> {
-          return (element.expiresAt() < Timer.getFPGATimestamp())
-              || (RobotConfig.get().perfToggles().noteMapBoundingBox()
-                  && safeToTrack()
-                  && noteInView(element.noteTranslation()));
+          return (element.expiresAt() < Timer.getFPGATimestamp()) || element.health() <= 10;
```

Comment:
this logic can be rewritten to be much more performant & simpler. here is how i would do it:
1. get a list of the remembered notes in the bounding box
2. loop through the list. for each element, remove the note from the array. if the health of the note was greater than 1, add it back to the array with `health - 1`

this logic generally seems okay but does a lot of extra work that isn't necessary:
1. by checking `RobotConfig.get().perfToggles().noteMapBoundingBox()` and `safeToTrack()` from within the `.filter()`, you are unnecessarily looping through the entire note array. both those conditions are global, they don't change depending on which note you're looking at
2. the removal logic is wrong, we remove notes once they are going to hit `health == 0`. the behavior here is to always remove them, since their health is always going to be `<= 10`
3. there are a few other things but none of that feedback is relevant as the updated logic will require rewriting that code anyway